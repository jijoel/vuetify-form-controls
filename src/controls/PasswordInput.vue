<template>
<v-text-field
  :value="value"
  v-bind="$attrs"
  v-on="listeners"
  @input="input"
  @blur="blur"
  :name="name"
  :label="label"
  :type="passwordType"
  :prepend-icon="prependIcon"
  :append-icon="passwordIcon"
  @click:append="show_password=!show_password"
  :browser-autocomplete="browserAutocomplete"
>
  <slot v-for="slot in Object.keys($slots)" :name="slot" :slot="slot"/>
  <template v-for="slot in Object.keys($scopedSlots)" :slot="slot" slot-scope="scope">
    <slot :name="slot" v-bind="scope"/>
  </template>
</v-text-field>
</template>

<script>
import ChildControl from '../mixins/ChildControl'

export default {

  name: 'PasswordInput',

  mixins: [ ChildControl ],

  props: {
    name: {
      type: String,
      default: 'password'
    },
    browserAutocomplete: {
      type: String,
      default: 'current-password'
    },
    hideIcon: {
      type: [Boolean, String],
      default: false
    },
    label: {
      type: String,
      default: 'Password'
    },
    prependIcon: {
      type: String,
      default: 'vpn_key'
    },
  },

  data: () => ({
    show_password: false,
  }),

  computed: {
      progress () {
        const x = Math.min(100, this.value.length * 10)
        console.log(x)
        return x
      },
      color () {
        return ['error', 'warning', 'success'][Math.floor(this.progress / 40)]
      },
    passwordIcon() {
      if (this.hideIcon)
        return ''

      return (this.show_password)
        ? 'visibility_off'
        : 'visibility'
    },
    passwordType() {
      return (this.show_password)
        ? 'text'
        : 'password'
    },
  },

}
</script>
