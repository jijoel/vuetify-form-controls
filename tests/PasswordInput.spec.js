import './setup'
import { common } from './common'
import { shallowMount } from '@vue/test-utils'
import PasswordInput from '../src/controls/PasswordInput'

describe('PasswordInput', () => {

  common(PasswordInput)

  test('icon can be hidden', () => {
    const wrapper = shallowMount(PasswordInput, {
      propsData: { 'hide-icon': true }
    })
    expect(wrapper.vm.passwordIcon).toBe('')
    expect(wrapper.attributes('append-icon')).toBe('')
    expect(wrapper.element).toMatchSnapshot()
  })

  test('icon and text field change when showing password', () => {
    const wrapper = shallowMount(PasswordInput)
    expect(wrapper.vm.passwordIcon).toBe('visibility')
    expect(wrapper.attributes('append-icon')).toBe('visibility')
    expect(wrapper.attributes('type')).toBe('password')

    wrapper.setData({ show_password: true })

    expect(wrapper.vm.passwordIcon).toBe('visibility_off')
    expect(wrapper.attributes('append-icon')).toBe('visibility_off')
    expect(wrapper.attributes('type')).toBe('text')
  })

})
