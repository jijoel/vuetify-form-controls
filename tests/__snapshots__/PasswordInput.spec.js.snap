// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput icon can be hidden 1`] = `
<v-text-field-stub
  append-icon=""
  browser-autocomplete="current-password"
  label="Password"
  name="password"
  prepend-icon="vpn_key"
  type="password"
  value=""
/>
`;

exports[`PasswordInput is rendered as custom vuetify text field 1`] = `
<v-text-field-stub
  append-icon="visibility"
  browser-autocomplete="current-password"
  label="Password"
  name="password"
  prepend-icon="vpn_key"
  type="password"
  value=""
/>
`;

exports[`PasswordInput renders properties sent by parent 1`] = `
<v-text-field-stub
  :solo="true"
  append-icon="visibility"
  browser-autocomplete="current-password"
  label="Password"
  name="password"
  prepend-icon="vpn_key"
  type="password"
  value=""
/>
`;
